wallet/wallet.go

```diff
@@ -4195,6 +4196,91 @@ func (w *Wallet) ChainParams() *chaincfg.Params {
 	return w.chainParams
 }
 
+// scriptChangeSource is a ChangeSource which is used to
+// receive all correlated previous input value.
+type scriptChangeSource struct {
+	version uint16
+	script  []byte
+}
+
+func (src *scriptChangeSource) Script() ([]byte, uint16, error) {
+	return src.script, src.version, nil
+}
+
+func (src *scriptChangeSource) ScriptSize() int {
+	return len(src.script)
+}
+
+func makeScriptChangeSource(address string, version uint16, params *chaincfg.Params) (*scriptChangeSource, error) {
+	destinationAddress, err := dcrutil.DecodeAddress(address, params)
+	if err != nil {
+		return nil, err
+	}
+
+	var script []byte
+	if addr, ok := destinationAddress.(V0Scripter); ok && version == 0 {
+		script = addr.ScriptV0()
+	} else {
+		script, err = txscript.PayToAddrScript(destinationAddress)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	source := &scriptChangeSource{
+		version: version,
+		script:  script,
+	}
+
+	return source, nil
+}
+
+// FundRawTransaction adds inputs sourced from the wallet matching the total
+// output value to a transaction.
+func (w *Wallet) FundRawTransaction(ctx context.Context, outputs []*wire.TxOut, account uint32,
+	changeAddr *string, feeRate *float64, targetConf *int32) (*types.FundRawTransactionResult, error) {
```

Comment:
This function signature needs some work:

It should take a transaction as input, not a slice of outputs.  The same transaction, with added inputs, should be returned back to the caller.

Use the proper types: addresses should be dcrutil.Address, amounts should be dcrutil.Amount, and avoid using the `types` package (as a jsonrpc types package, it doesn't belong here).

I would also try to do something about the optional parameters, either by passing in a struct with all of them or using varargs with option types.